x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -716,15 +715,41 @@ ActionListener<Void> addCompletionInfo(
         });
     }
 
-    static PhysicalPlan reductionPlan(ExchangeSinkExec plan, boolean enable) {
-        PhysicalPlan reducePlan = new ExchangeSourceExec(plan.source(), plan.output(), plan.isIntermediateAgg());
-        if (enable) {
-            PhysicalPlan p = PlannerUtils.reductionPlan(plan);
-            if (p != null) {
-                reducePlan = p.replaceChildren(List.of(reducePlan));
-            }
+    static ReductionPlan reductionPlan(
+        PlannerSettings plannerSettings,
+        EsqlFlags flags,
+        Configuration configuration,
+        FoldContext foldCtx,
+        ExchangeSinkExec originalPlan,
+        boolean runNodeLevelReduction,
+        boolean splitTopN
+    ) {
+        PhysicalPlan source = new ExchangeSourceExec(originalPlan.source(), originalPlan.output(), originalPlan.isIntermediateAgg());
+        if (splitTopN == false && runNodeLevelReduction == false) {
```

Comment:
Shouldn't that just be
```suggestion
        if (runNodeLevelReduction == false) {
```
?

I guess the case where `runNodeLevelReduction == false` and `splitTopN == true` is invalid to begin with, but if we were to end up in this case, the current code will happily apply a node-level reduction for TopN even when it's disabled per `runNodeLevelReduction`, no?